src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
+import usePortal from 'react-cool-portal';
+import styled from 'styled-components';
+import { isNotNull, isNotUndefined } from 'ramda-adjunct';
+
+import { ModalProps } from './Modal.types';
+import { ModalWidthSizes } from './Modal.enums';
+import { useCalculateContentHeight } from './hooks/useCalculateContentHeight';
+import { useLockBodyScroll } from './hooks/useLockBodyScroll';
+import { useOuterClick } from '../../hooks/useOuterCallback';
+import { Icon } from '../Icon';
+import { Padbox } from '../layout';
+import { H3 } from '../typography';
+import { getBorderRadius, getColor, getFontSize, pxToRem } from '../../utils';
+import { mergeRefs } from '../../utils/mergeRefs';
+import { SpaceSizes } from '../../theme';
+import { DSContext } from '../../theme/DSProvider/DSProvider';
+import { IconTypes, SSCIconNames } from '../../theme/icons/icons.enums';
+
+const widthVariants = {
+  [ModalWidthSizes.xs]: 320,
+  [ModalWidthSizes.sm]: 430,
+  [ModalWidthSizes.md]: 600,
+  [ModalWidthSizes.lg]: 900,
+};
+
+const Overlay = styled.div`
+  position: fixed;
+  width: 100%;
+  height: 100%;
+  top: 0;
+  background: rgba(0, 0, 0, 0.6);
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  z-index: 1250;
+`;
+
+const BaseModal = styled.div<{ $width: number }>`
+  position: relative;
+  width: ${({ $width }) => pxToRem($width)};
+  border-radius: ${getBorderRadius};
+  background-color: ${getColor('graphite5H')};
+`;
+
+const Title = styled(H3)`
+  margin: 0;
+`;
+
+// TODO Extract close button and unify it across DS
+const CloseButton = styled.button`
```

Comment:
I personally would put some padding in this button so we increase the interactive area so it can be easily used with touchscreens

Reply:
We discussed with Olya the size and it should be at least 24x24. In modal it is fine as the button has absolute position but I can't  extract it and reuse it in Banner, it would break the layout there. We should curate the AC for close button like sizes, color change on hover etc. Anyway, I will increase the padding here